apps/prairielearn/src/pages/instructorCourseAdminSettings/instructorCourseAdminSettings.js

```diff
@@ -1,53 +1,41 @@
 //@ts-check
 import * as express from 'express';
 import * as path from 'path';
-const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'));
+const asyncHandler = require('express-async-handler');
 import * as fs from 'fs-extra';
-import * as async from 'async';
 const ERR = require('async-stacktrace');
 import { CourseInfoEditor } from '../../lib/editors';
 import { logger } from '@prairielearn/logger';
 import * as error from '@prairielearn/error';
 
 const router = express.Router();
 
-router.get('/', function (req, res, next) {
-  debug('GET /');
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    try {
+      await fs.access(res.locals.course.path);
+    } catch (err) {
+      if (err.code === 'ENOENT') {
+        res.locals.needToSync = true;
+      } else {
+        throw new Error(err);
+      }
+    }
```

Comment:
Now that I see this written out, I think using [`pathExists` from `fs-extra`](https://github.com/jprichardson/node-fs-extra/blob/master/docs/pathExists.md) is probably the cleaner solution. I asked for opinions about this on Slack; we'll see if anyone feels strongly either way. Hold off on changing this for right now!

Reply:
As discussed on Slack, we'll go with `pathExists` here.